scripts/promise_batch_imports.py

```diff
@@ -56,8 +59,12 @@ def map_book_to_olbook(book, promise_id):
         'authors': [{"name": book['ProductJSON'].get('Author') or '????'}],
         'publishers': [book['ProductJSON'].get('Publisher') or '????'],
         'source_records': [f"promise:{promise_id}:{sku}"],
-        # format_date adds hyphens between YYYY-MM-DD
-        'publish_date': publish_date and format_date(publish_date) or '????',
+        # format_date adds hyphens between YYYY-MM-DD, or use only YYYY if date is suspect.
+        'publish_date': format_date(
+            date=publish_date, only_year=publish_date[-4:] in ('0000', '0101')
+        )
+        if publish_date
+        else '????',
```

Comment:
@cdrini These `????` look wrong and are causing problems when reusing OL date data, if there is no known date the field should not be set.

Reply:
I think these are handled later in the process and the `????` are correctly prevented from being written to OL.

Reply:
I believe that's right :+1: But not an expert on this code!

Reply:
@hornc, sorry for missing this for so long. The issue of `????` as a `publish_date` should be addressed by #7518, which was committed back in March, so hopefully since then there have been no `publish_date`s of `????`.